### DISCUSSION TITLE
Disabling the form as a whole 😅?

### DISCUSSION THREAD

**Discussion by Huiet** [mantinedev/mantine#2794] [Q&A] [answered]
Currently in order to disable my form I have used a context and wrapped each input in a component that consumes this context and determines whether it is disabled or not. (I do something similar to decide whether the inputs are visible or not).

This is ok but pretty verbose; Is there any appetite to integrate something like a make read only or disable to the form as a whole. Or, if we wanted to be real fancy (this is effectively what I did), make it so you can pass a list of props that would then be disabled. 

It could be way more gross than it seems on the surface, but perhaps tying a disabled prop to getinputprops could get that functionality in place in a somewhat elegant way. It should be backwards compatible with only the order possibly needing to change to fully leverage it in some projects.

**Comment by rtivital** [comment, marked as answer]
I do not think that use-form should manage disabled prop of the inputs – hook purpose is to work with data and validation. Disabled and readOnly props are on the UI side of things – they should be set as component props.
